src/Responses/Responses/Tool/RemoteMcpTool.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenAI\Responses\Responses\Tool;
+
+use OpenAI\Contracts\ResponseContract;
+use OpenAI\Responses\Concerns\ArrayAccessible;
+use OpenAI\Testing\Responses\Concerns\Fakeable;
+
+/**
+ * @phpstan-type RemoteMcpToolType array{type: 'mcp', server_label: string, server_url: string, require_approval: 'never'|null, allowed_tools: string[]|null, headers: array<string, string>|null}
+ *
+ * @implements ResponseContract<RemoteMcpToolType>
+ */
+final class RemoteMcpTool implements ResponseContract
+{
+    /**
+     * @use ArrayAccessible<RemoteMcpToolType>
+     */
+    use ArrayAccessible;
+
+    use Fakeable;
+
+    /**
+     * @param  'mcp'  $type
+     * @param  'never'|null  $requireApproval
+     * @param  string[]|null  $allowedTools
+     * @param  array<string, string>|null  $headers
+     */
+    private function __construct(
+        public readonly string $type,
+        public readonly string $serverLabel,
+        public readonly string $serverUrl,
+        public readonly ?string $requireApproval = null,
```

Comment:
Looking at docs `allowed_tools` I'm a bit confused. Its an array of strings OR an object (MCPAllowedToolFilter) that has an array of tool names? It seems we only support the array notation.

Reply:
<img width="474" alt="Screenshot 2025-06-13 at 10 47 50 AM" src="https://github.com/user-attachments/assets/64615fa6-74aa-4e6b-bbae-e7324dbb795c" />